hashicorp/consul
# [CC-7041] Update and start the SCADA provider in HCP manager

### Description
This PR starts the SCADA provider in the HCP manager after updating it with the HCP configuration. We want to be less dependent on using the HCP configuration at startup so that we can eventually support updating the HCP config at runtime. The SCADA provider still needs to always be initialized at startup, though, since its listener needs to be registered with the Consul server. 

This PR is best viewed commit-by-commit, where the changes are:

1. Bump the SCADA provider version so that it has support for updating the configuration after initialization.
2. Use a placeholder HCP config when initializing the SCADA provider and then after initialization, update with the actual HCP config immediately after for now.
3. Save the entire cloud configuration as a dependency rather than just the management token and pass it to the HCP manager.
4. Move updating and starting the SCADA provider into the HCP manager.

### Testing & Reproduction steps
Manually E2E tested this with the following steps:

1. Configure and start Consul with a cloud configuration.
2. Inspect that the SCADA connection requests are succeeding in the Consul logs.
3. Inspect in HCP Consul Central that the cluster is in a RUNNING state and that basic functionality works (services details are accurate, hosted UI works, cluster peering works) to ensure the SCADA connection is working properly.

```
2023-12-18T11:39:30.979-0600 [DEBUG] agent.hcp_manager: HCP manager starting
2023-12-18T11:39:30.979-0600 [DEBUG] agent.hcp_manager: updating scada provider with HCP configuration
2023-12-18T11:39:30.979-0600 [INFO]  agent.hcp.scada.scada-provider: starting
2023-12-18T11:39:30.982-0600 [INFO]  agent: Starting server: address=scada::core_api network=scada protocol=http
2023-12-18T11:39:31.373-0600 [DEBUG] agent.hcp.scada.scada-provider: assigned session ID: id=cfc0a73f-362d-e00e-cc00-7a431c814856
2023-12-18T11:39:34.680-0600 [DEBUG] agent.hcp.scada.scada-provider: accepted connection
2023-12-18T11:39:34.681-0600 [DEBUG] agent.hcp.scada.scada-provider: connect requested: capability=core_api
2023-12-18T11:39:53.371-0600 [DEBUG] agent.hcp.scada.scada-provider: accepted connection
```

### Links
- Jira: https://hashicorp.atlassian.net/browse/CC-7041

### PR Checklist
